include/drivers/uart.h

```diff
@@ -551,8 +551,9 @@ static inline int uart_rx_buf_rsp(struct device *dev, u8_t *buf, size_t len)
 /**
  * @brief Disable RX
  *
- * UART_RX_BUF_RELEASED event will be generated for every buffer scheduled,
- * after that UART_RX_DISABLED event will be generated.
+ * UART_RX_RDY will be generated if there is pending received data followed by
+ * UART_RX_BUF_RELEASED event for every buffer scheduled, after that
```

Comment:
This should be rephrased. Currently this "pending received data followed by UART_RX_BUF_RELEASED event" is quite confusing.
In addition to the mention here, the description of the [UART_RX_RDY](https://github.com/zephyrproject-rtos/zephyr/blob/3d71e790b9acb6d2f3ea0b5694211f232444221a/include/drivers/uart.h#L99) event should be updated (currently it mentions only two cases in which this event is generated). Also the "Receiving" section, in part that [describes](https://github.com/zephyrproject-rtos/zephyr/blob/3d71e790b9acb6d2f3ea0b5694211f232444221a/include/drivers/uart.h#L67) the UART_RX_STOPPED and UART_RX_BUF_RELEASED events, could mention that before a given buffer is released, the UART_RX_RDY event is generated for any data received to that buffer that has not yet been passed to the user callback.

Reply:
can you take another look?

Reply:
I still find it more confusing than explaining. What about just adding a sentence about the UART_RX_RDY event after the description that we've got now? I mean something like:
```
 * Additionally, if there is any pending received data, the UART_RX_RDY event
 * for that data will be generated before the UART_RX_BUF_RELEASED events.
```